include/boost/beast/http/impl/fields.hpp

```diff
@@ -548,9 +548,24 @@ void
 basic_fields<Allocator>::
 insert(string_view sname, string_view const& value)
 {
-    auto const name =
-        string_to_field(sname);
-    insert(name, sname, value);
+    insert(
+        string_to_field(sname), sname, value);
+}
+
+template<class Allocator>
+void
+basic_fields<Allocator>::
+insert(
+    field name,
+    string_view sname,
+    string_view const& value,
```

Comment:
this is weird, why is this passed by reference and the other passed by value?

Reply:
I didn't modify the existing signatures. Yes, they are strange, and we need to address them all.